File: src/decoders/timetable-class-activity.ts
import type { SessionHandle, TimetableClassActivity } from "~/models";
import { decodeTimetableClass } from "./timetable-class";

export const decodeTimetableClassActivity = (item: any, session: SessionHandle): TimetableClassActivity => {
  return {
    is: "activity",
    ...decodeTimetableClass(item, session),

    title: item.motif,
    attendants: item.accompagnateurs,
    resourceTypeName: item.strGenreRess,
    resourceValue: item.strRess
  };
};

File: library/src/commonMain/kotlin/decoders/timetable-class-activity.kt
package decoders

import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.jsonArray
import kotlinx.serialization.json.jsonPrimitive
import models.SessionInformation
import models.TimetableClassActivity

fun decodeTimetableClassActivity (item: JsonObject, sessionInformation: SessionInformation) = TimetableClassActivity(
    title = item["motif"]!!.jsonPrimitive.content,
    attendants = item["accompagnateurs"]!!.jsonArray.map { it.jsonPrimitive.content },
    resourceTypeName = item["strGenreRess"]!!.jsonPrimitive.content,
    resourceValue = item["strRess"]!!.jsonPrimitive.content
)